source/common/http/async_client_impl.cc

```diff
@@ -16,22 +16,20 @@ namespace Envoy {
 namespace Http {
 
 const std::vector<std::reference_wrapper<const Router::RateLimitPolicyEntry>>
-    AsyncStreamImpl::NullRateLimitPolicy::rate_limit_policy_entry_;
-const AsyncStreamImpl::NullHedgePolicy AsyncStreamImpl::RouteEntryImpl::hedge_policy_;
-const AsyncStreamImpl::NullRateLimitPolicy AsyncStreamImpl::RouteEntryImpl::rate_limit_policy_;
-const Router::InternalRedirectPolicyImpl AsyncStreamImpl::RouteEntryImpl::internal_redirect_policy_;
-const Router::PathMatcherSharedPtr AsyncStreamImpl::RouteEntryImpl::path_matcher_;
-const Router::PathRewriterSharedPtr AsyncStreamImpl::RouteEntryImpl::path_rewriter_;
-const std::vector<Router::ShadowPolicyPtr> AsyncStreamImpl::RouteEntryImpl::shadow_policies_;
-const AsyncStreamImpl::NullVirtualHost AsyncStreamImpl::RouteEntryImpl::virtual_host_;
-const AsyncStreamImpl::NullRateLimitPolicy AsyncStreamImpl::NullVirtualHost::rate_limit_policy_;
-const AsyncStreamImpl::NullCommonConfig AsyncStreamImpl::NullVirtualHost::route_configuration_;
-const std::multimap<std::string, std::string> AsyncStreamImpl::RouteEntryImpl::opaque_config_;
-const AsyncStreamImpl::NullPathMatchCriterion
-    AsyncStreamImpl::RouteEntryImpl::path_match_criterion_;
-const AsyncStreamImpl::RouteEntryImpl::ConnectConfigOptRef
-    AsyncStreamImpl::RouteEntryImpl::connect_config_nullopt_;
-const std::list<LowerCaseString> AsyncStreamImpl::NullCommonConfig::internal_only_headers_;
+    NullRateLimitPolicy::rate_limit_policy_entry_;
+const NullHedgePolicy RouteEntryImpl::hedge_policy_;
+const NullRateLimitPolicy RouteEntryImpl::rate_limit_policy_;
+const Router::InternalRedirectPolicyImpl RouteEntryImpl::internal_redirect_policy_;
+const Router::PathMatcherSharedPtr RouteEntryImpl::path_matcher_;
+const Router::PathRewriterSharedPtr RouteEntryImpl::path_rewriter_;
+const std::vector<Router::ShadowPolicyPtr> RouteEntryImpl::shadow_policies_;
+const NullVirtualHost RouteEntryImpl::virtual_host_;
+const NullRateLimitPolicy NullVirtualHost::rate_limit_policy_;
+const NullCommonConfig NullVirtualHost::route_configuration_;
+const std::multimap<std::string, std::string> RouteEntryImpl::opaque_config_;
+const NullPathMatchCriterion RouteEntryImpl::path_match_criterion_;
+const RouteEntryImpl::ConnectConfigOptRef RouteEntryImpl::connect_config_nullopt_;
+const std::list<LowerCaseString> NullCommonConfig::internal_only_headers_;
```

Comment:
nit: I believe these are all the static global variables associated with the classes in null_route_impl.h. Let's create null_route_impl.cc and move them there.

Reply:
done